pkg/auth/providers/oidc/oidc_actions.go

```diff
@@ -42,7 +42,7 @@ func (o *OpenIDCProvider) ActionHandler(actionName string, action *types.Action,
 func (o *OpenIDCProvider) ConfigureTest(actionName string, action *types.Action, request *types.APIContext) error {
 	//verify body has all required fields
 	input, err := handler.ParseAndValidateActionBody(request, request.Schemas.Schema(&managementschema.Version,
-		o.getType()))
+		o.Type))
```

Comment:
So this all gets inherited by keycloak OIDC ? And therefore O is of keycloak type.

Reply:
Yes, the `o` at this point is the `OpenIDCProvider` on the `keyCloakOIDCProvider` struct. This issue that this fixes was  was when functions in the OpenIDCProvider interface were being called, it had no idea about the "overriden"  `o.getType()` that I had created in `keyCloakOIDCProvider` so it wasn't returning OpenIDCProvider values and not the keyCloakOIDCProvder values. This meant some types and names were getting set wrong ultimately leading to both providers getting enabled with the rest of the problems that come with it. By setting the value as string, it ensures the value will be available regardless of if its called in the interface or if I change the func definition in the `keyCloakOIDCPRovider` struct. There could be better ways to resolve this, but this seemed like a good solution for now.

Reply:
I had to brush up my memory here to remember: https://play.golang.org/p/IGMYjG9mQbQ

> Embedding types introduces the problem of name conflicts but the rules to resolve them are simple. First, a field or method X hides any other item X in a more deeply nested part of the type. If log.Logger contained a field or method called Command, the Command field of Job would dominate it.

https://golang.org/doc/effective_go#embedding

If you added a keycloak.printer there it would only return that one. So in reverse if you dropped the GetType from Keycloak it would refer back to the OIDC one. 

Using strings avoids this whole confusion.